bitnami/influxdb/values.yaml

```diff
@@ -960,77 +986,63 @@ backup:
-      ## @param backup.uploadProviders.google.image.tag Google Cloud SDK image tag
-      ## @param backup.uploadProviders.google.image.pullPolicy Google Cloud SDK image pull policy
-      ## @param backup.uploadProviders.google.image.pullSecrets Specify docker-registry secret names as an array
-      ##
-      image:
-        registry: docker.io
-        repository: bitnami/google-cloud-sdk
-        tag: 0.365.0-debian-10-r11
-        ## Specify a imagePullPolicy. Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
-        ## ref: https://kubernetes.io/docs/user-guide/images/#pre-pulling-images
-        ##
-        pullPolicy: IfNotPresent
-        ## Optionally specify an array of imagePullSecrets.
-        ## Secrets must be manually created in the namespace.
-        ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
-        ## e.g:
-        ## pullSecrets:
-        ##   - myRegistryKeySecretName
-        ##
-        pullSecrets: []
-    ## Azure Storage Container configuration
-    ##
-    azure:
-      ## @param backup.uploadProviders.azure.enabled Enable upload to azure storage container
-      ## @param backup.uploadProviders.azure.secret Secret with credentials to access Azure storage
-      ## @param backup.uploadProviders.azure.secretKey Service account secret key name
-      ## @param backup.uploadProviders.azure.existingSecret Name of existing secret object
-      ## @param backup.uploadProviders.azure.containerName Destination container
+      account: ""
+      blobKey: ""
+      sasUrl: ""
+      endpoint: ""
+    ## ref: https://rclone.org/googlecloudstorage/
+    googlecloudstorage:
+      ## @param backup.uploadProviders.googlecloudstorage.enabled Enable Google Cloud Storage
+      ## @param backup.uploadProviders.googlecloudstorage.clientId OAuth client ID
+      ## @param backup.uploadProviders.googlecloudstorage.clientSecret OAuth client secret
+      ## @param backup.uploadProviders.googlecloudstorage.projectNum Project number
+      ## @param backup.uploadProviders.googlecloudstorage.objectACL Access control list for new objects
+      ## @param backup.uploadProviders.googlecloudstorage.bucketACL Access control list for new buckets
+      ## @param backup.uploadProviders.googlecloudstorage.location Location for newly created buckets
+      ## @param backup.uploadProviders.googlecloudstorage.storageClass The storage class to use when storing in Googe Cloud Storage
+      ## @param backup.uploadProviders.googlecloudstorage.token OAuth access token as a JSON blob
+      ## @param backup.uploadProviders.googlecloudstorage.authUrl Auth server URL
+      ## @param backup.uploadProviders.googlecloudstorage.tokenUrl Token server URL
+      enabled: false
+      clientId: ""
+      clientSecret: ""
+      projectNum: ""
+      objectACL: ""
+      bucketACL: ""
+      location: ""
+      storageClass: ""
+      token: ""
+      authUrl: ""
+      tokenUrl: ""
```

Comment:
You're using all these parameters to construct the `uploads.sh` script where all these credentials will be in plaintext. This script is included in the ConfigMap, therefore it's not secure at all.
I'd suggest to create a YAML file **in a Secret** including all these GCS details. This'll help you to simplify the values structure also, because you'll be able to use an approach like the below in the Secret:
```yaml
data:
  {{- if .Values.backup.uploadProviders.googlecloudstorage.enabled }}
  gcs.yaml: {{- omit .Values.backup.uploadProviders.googlecloudstorage "enabled" | toYaml | b64enc | quote }}
  {{- end }}
```
Using this approach any parameter under the `backup.uploadProviders.googlecloudstorage` map (but the "enabled" one) will be included. This way, you can remove any non-mandatory property from this map in the **values.yaml** structure.

Reply:
This recommendation can be extended to other cloud providers.

Reply:
Thanks for the review. I'll try to get this done